File: exercism/python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = [num for num in matrix_string.split("\n")]

    def row(self, index):
        return [int(num) for num in self.matrix[index - 1].split() if num.isdigit()]

    def column(self, index):
        total = [numbers.split(" ") for numbers in self.matrix]
        return [int(number[index - 1]) for number in total]

File: exercism/javascript/matrix/matrix.js
export class Matrix {
	constructor(string) {
		this.rows = this.processRows(string.split('\n'));
		this.columns = this.processColumns(this.rows);
	}

	processRows(numbers) {
		return numbers.map((num) => {
			let row = num.split(' ');

			let rowAsInt = row.map((int) => {
				return parseInt(int);
			});
			return rowAsInt;
		});
	}

	processColumns(rows) {
		const eachColumn = [];

		let i = 0;
		while (i <= rows.length) {
			let numberAtIndex = rows.map((row) => {
				return row[i];
			});
			eachColumn.push(numberAtIndex);
			i += 1;
		}
		return eachColumn;
	}
}
